### PR TITLE
Ignore disabled Rabbits in NnfSystemStorage

### PR DESCRIPTION
**Pull request by matthew-richerson**
If a Rabbit is marked as "Disabled" in the DWS storage resource, add an option to exlude it from the NnfSystemStorage Rabbit list.

Also, the Storage resource is created in DWS now, so remove the creation in nnf-sos. Add Watch()'s for various resources so that the NnfSystemStorage automatically adds storage on Rabbits that transition from "Disabled" to "Enabled".

### CODE REVIEW

File: internal/controller/nnfsystemstorage_controller.go [hunk 1]
```diff
@@ -254,6 +257,40 @@ func (r *NnfSystemStorageReconciler) createServers(ctx context.Context, nnfSyste
...
+			labels := storage.GetLabels()
+			if labels == nil {
+				continue
+			}
+
+			if storageType := labels[dwsv1alpha2.StorageTypeLabel]; storageType != "Rabbit" {
+				continue
+			}
+
+			if storage.Spec.State == dwsv1alpha2.DisabledState || storage.Status.Status == dwsv1alpha2.DisabledStatus {
```
Comment (roehrich-hpe) [inline internal/controller/nnfsystemstorage_controller.go:285]: Should we instead skip anything that has a status != ReadyStatus?

**Review by ajfloeder — APPROVED** [review]

**Review by roehrich-hpe — APPROVED** [review]

File: api/v1alpha1/conversion.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ package v1alpha1
 
 import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
```
[the remark this answers is not in the document]
Comment (matthew-richerson) [reply api/v1alpha1/conversion.go:24]: That's a different conversion.go file. I just copied from the dws conversion.go file.

File: internal/controller/nnf_access_controller.go [hunk 3]
```diff
@@ -1086,6 +1148,7 @@ func (r *NnfAccessReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(controller.Options{MaxConcurrentReconciles: maxReconciles}).
 		For(&nnfv1alpha2.NnfAccess{}).
+		Watches(&dwsv1alpha2.Computes{}, handler.EnqueueRequestsFromMapFunc(r.ComputesEnqueueRequests)).
```
Comment (bdevcich) [inline internal/controller/nnf_access_controller.go:1151]: so this affectively triggers a reconcile with the NnfAccess controller?

Comment (matthew-richerson) [reply internal/controller/nnf_access_controller.go:1151]: Yes, if the Computes resource for an NnfSystemStorage (or a Workflow) changes, the associated NnfAccess will reconcile again. For the Workflow, the Computes resource won't change, so this really only affects NnfSystemStorage.

**Review by bdevcich — APPROVED** [review]

**Pull request merged**